src/route/index.ts

```diff
@@ -545,17 +547,37 @@ export async function initApiServer(
             const simulationResponse = (await server.simulateTransaction(
               tx
             )) as SorobanRpc.Api.SimulateTransactionSuccessResponse;
+
             const preparedTransaction = SorobanRpc.assembleTransaction(
               tx,
               simulationResponse
             );
+
+            const built = preparedTransaction.build();
+            const sorobanOp = built
+              .operations[0] as Operation.InvokeHostFunction;
```

Comment:
should we inspect all operations or is only looking at the first one always sufficient?

Reply:
In the case of a tx with an InvokeHostFunction, there can only be 1 operation. We could check for the operation type here but only the InvokeHostFunction type has auth. I'll add an explicit check for the op type though, that seems a bit more clear.

Reply:
Added an explicit check for op type in 3272c3c01df00ef3db048298a44055acca7e0b97